src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -0,0 +1,251 @@
+
+  /** The checked state of the radio button. */
+  @Prop({ mutable: true, reflect: true }) checked: boolean = false;
+  @Watch("checked")
+  onCheckedChange(newChecked: boolean, oldChecked: boolean) {
+    if (newChecked === true && oldChecked === false) {
+      this.uncheckOtherRadioButtonsInGroup();
+    }
+    if (newChecked !== this.input.checked) {
+      this.input.checked = newChecked;
+    }
+    this.calciteRadioButtonChange.emit();
+  }
+
+  /** The disabled state of the radio button. */
+  @Prop({ reflect: true }) disabled?: boolean = false;
+  @Watch("disabled")
+  onDisabledChange(disabled: boolean) {
+    this.input.disabled = disabled;
+  }
+
+  /** The focused state of the radio button. */
+  @Prop({ mutable: true, reflect: true }) focused: boolean = false;
+  @Watch("focused")
+  onFocusedChange(focused: boolean) {
+    if (focused) {
+      this.input.focus();
+    } else {
+      this.input.blur();
+    }
+  }
+
+  /** The id attribute of the radio button.  When omitted, a globally unique identifier is used. */
+  @Prop({ reflect: true }) guid: string =
+    this.el.id || `calcite-radio-button-${guid()}`;
+
+  /** The name of the radio button.  <code>name</code> is passed as a property automatically from <code><calcite-radio-button-group></code>. */
+  @Prop({ reflect: true }) name!: string;
+  @Watch("name")
+  onNameChange(newName: string) {
+    if (newName !== this.input.name) {
+      this.input.name = newName;
+    }
+  }
+
+  /** Requires that a value is selected for the radio button group before the parent form will submit. */
+  @Prop({ reflect: true }) required: boolean = false;
+  @Watch("required")
+  onRequiredChange(required: boolean) {
+    this.input.required = required;
+  }
+
+  /** The scale (size) of the radio button.  <code>scale</code> is passed as a property automatically from <code><calcite-radio-button-group></code>. */
+  @Prop({ mutable: true, reflect: true }) scale: "s" | "m" | "l" = "m";
+  @Watch("scale")
+  validateScale(newScale: string) {
+    const scales = ["s", "m", "l"];
+    if (!scales.includes(newScale)) this.scale = "m";
+  }
+
+  /** The color theme of the radio button, <code>theme</code> is passed as a property automatically from <code><calcite-radio-button-group></code>. */
+  @Prop({ mutable: true, reflect: true }) theme: "light" | "dark" = "light";
+  @Watch("theme")
+  validateTheme(newTheme: string) {
+    const themes = ["light", "dark"];
+    if (!themes.includes(newTheme)) this.theme = "light";
+  }
+
+  /** The value of the radio button. */
+  @Prop() value!: string;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Properties & Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private input: HTMLInputElement;
+
+  private checkFirstRadioButton = () => {
+    let radioButtons = document.querySelectorAll(
+      `calcite-radio-button[name=${this.name}]`
```

Comment:
Maybe it's an edge case, but wouldn't this run into issues if you had two forms and each one had buttons with same names?

Reply:
Correct.  I thought of this as well so maybe one way to address this is to allow `radio-button` to be aware of its parent `<form>` in some way so that it can execute a more refined query.  Any suggestions welcome on that, but like you said it's likely a rare edge case.